### PR TITLE
prune elvish

### PR DESCRIPTION
**Pull request by rminnich**
This is an attempt to shrink and simplify elvish.
Consider it a checkpoint commit.

It has impact in the sense of cutting down elvish complexity
dramatically, but so far I don't see huge impact on size:
compressed images

-rwxr-xr-x 1 rminnich rminnich 15252220 Oct 6 10:04 /tmp/initramfs.linux_amd64.cpio
-rwxr-xr-x 1 rminnich rminnich 5695727 Oct 6 10:04 /tmp/initramfs.linux_amd64.cpio.gz

-rwxr-xr-x 1 rminnich rminnich 15022780 Oct 6 10:03 /tmp/smallelvish.cpio
-rwxr-xr-x 1 rminnich rminnich 5610432 Oct 6 10:03 /tmp/smallelvish.cpio.gz

so only 200K or so uncompressed and 80k compressed.

OTOH the stripped elvish does shrink from 4.7 to 3.9M so that's something -- almost 20%.

At the same time I may be on the edge of a lot of removal which will shrink it greatly,
but it's hard to tell. There's lots of stuff in elvish that seems to come from a void*
world (e.g. prolific use of interface{}, all the hashmap stuff)
and it would be nice to make that disapperar

But we do remove JSON from elvish, and we do remove all of xiaq's vendored packages,
and we do remove use of unsafe, and those may have value in and of themselves.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rjoleary** [comment]
I'll debug why the gotest integration test failed. For now, you can rebase without it. The serial output for the integration test are here: https://18770-46462030-gh.circle-artifacts.com/0/go/src/github.com/u-root/u-root/integration/serial/gotest.log

File: cmds/elvish/hash/hash.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func DJB(hs ...uint32) uint32 {
+	acc := DJBInit
+	for _, h := range hs {
+		acc = DJBCombine(acc, h)
+	}
+	return acc
+}
+
+func Hash(i interface{}) uint32 {
+	s := fmt.Sprintf("%x", i)
```
[the remark this answers is not in the document]
Comment (GanShun) [reply cmds/elvish/hash/hash.go]: I just realized you moved this from outside

**Comment by GanShun** [comment]
Ok this seems to work for me

**Comment by rjoleary** [comment]
The gotest is passing now too.

**Pull request merged**